flake.nix

```diff
@@ -17,22 +20,67 @@
     };
   };
 
-  outputs = { self, nixpkgs, flake-utils, cardano-node }:
+  outputs = { self, nixpkgs, flake-utils, cardano-node, poetry2nix }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
           pkgs = nixpkgs.legacyPackages.${system};
-          python3 = pkgs.python3.override {
-            packageOverrides = self: _: {
-              allure = self.callPackage ./nix/allure.nix { };
-              pytest-allure = self.callPackage ./nix/pytest-allure.nix { };
-              pytest-select = self.callPackage ./nix/pytest-select.nix {};
-              cardano-clusterlib = self.callPackage ./nix/cardano-clusterlib.nix { };
-            };
+          p2n = poetry2nix.legacyPackages.${system};
+          # Workarounds `ModuleNotFoundError: No module named 'poetry'`
+          # cf. https://github.com/nix-community/poetry2nix/blob/master/docs/edgecases.md#modulenotfounderror-no-module-named-packagename
+          pypkgs-build-requirements = {
+            pytest-metadata = [ "poetry" ];
+            pytest-select = [ "poetry" ];
+            m2r2 = [ "setuptools" ];
+            sphinxemoji = [ "setuptools" ];
+            sphinxcontrib-jquery = [ "flit-core" ];
+          };
+          p2n-overrides = p2n.defaultPoetryOverrides.extend (self: super:
```

Comment:
```suggestion
          # for the time being we need some minor adjustments to the constructed poetry project
          # in nix.
          #
          # 1. poetry2nix fails to see the setuptools-scm dependency, hence we need to inject it by hand into Cardano-cluster lib.
          # 2. Because transitively depend on `py`, (through `pytest-html`), we need to drop the module from `pytest`.
          # 
          p2n-overrides = p2n.defaultPoetryOverrides.extend (self: super:
```